src/shared/acl/acl.service.ts

```diff
@@ -24,26 +27,32 @@ export class BaseAclService {
    */
   public forUser = (user: { id: number; roles: ROLE[] }): any => {
     return {
-      canDoAction: (action: Action, subject: Subject) => {
+      canDoAction: (action: Action, subject?: Subject) => {
         let canDoAction = false;
+
         user.roles.forEach((userRole) => {
           //If already has accress, return
           if (canDoAction) return true;
 
-          //find acl rule for given user role
-          const aclRule = this.acl.find((rule) => rule.role === userRole);
+          //find all rules for given user role
+          const aclRules = this.acl.filter((rule) => rule.role === userRole);
+
+          //for each rule, check action permission
+          aclRules.forEach((aclRule) => {
+            //If already has accress, return
+            if (canDoAction) return true;
 
-          //if acl rule found, check action permission
-          const hasActionPermission =
-            aclRule &&
-            (aclRule.actions.includes(action) ||
-              aclRule.actions.includes(Action.Manage));
+            //check action permission
+            const hasActionPermission =
+              aclRule.actions.includes(action) ||
+              aclRule.actions.includes(Action.Manage);
 
-          // check resource ownership rule
-          canDoAction =
-            hasActionPermission &&
-            (!aclRule.isResourceOwner ||
-              aclRule.isResourceOwner(subject, user));
+            //check for custom `isResourceOwner` callback
+            canDoAction =
+              hasActionPermission &&
+              (!aclRule.isResourceOwner ||
```

Comment:
```suggestion
              (aclRule.isResourceOwner &&
```
Would this be correct?